### DISCUSSION TITLE
Filter active silences with regex

### DISCUSSION THREAD

**Discussion by grimz-ly** [prometheus/alertmanager#3665] [Q&A]
If I query a silence through the api I notice "isRegex":false being returned if it was created on an exact label value.

Does this mean there is no way to query with regex and include the silence?

looking to match host=~dcc-.* and get hundreds of silences returned with this hostname prefix that were created in order to expire them all.

```curl -g 'http://alertmanager:9093/api/v2/silences?filter=host=~"dcc-.*"'```

returns nothing.

**Comment by caislen42** [comment]
I think that isRegex only relates to how the silence is used to match alerts, not how you can filter silences in the api. 

That said, I have a similar issue, being unable to use regex queries after upgrading from 0.21.0. Previously filter=host=~"A|B|C" worked fine for listing silences belonging to all of the hosts, but with 0.-22.0 only empty response.

**Comment by TheMeier** [comment]
The filter content is a matcher that is used in an _string_ equal comparison https://github.com/prometheus/alertmanager/blob/main/api/v2/api.go#L629

IMHO it would be kind of tricky to apply a reg regex matcher with another regex. Eg matcher is `foo=~"bar.*"` should something like this be possible then `filter="fo.*=\"ba.*\""? Should it match the `.*` inside the matcher?
